implementation/src/main/java/io/smallrye/mutiny/groups/MultiOnTerminate.java

```diff
@@ -0,0 +1,75 @@
+package io.smallrye.mutiny.groups;
+
+import static io.smallrye.mutiny.helpers.ParameterValidation.nonNull;
+
+import java.util.function.BiConsumer;
+import java.util.function.BiFunction;
+
+import io.smallrye.mutiny.Multi;
+import io.smallrye.mutiny.Uni;
+import io.smallrye.mutiny.helpers.ParameterValidation;
+import io.smallrye.mutiny.infrastructure.Infrastructure;
+import io.smallrye.mutiny.operators.multi.MultiOnTerminationInvokeUni;
+import io.smallrye.mutiny.operators.multi.MultiSignalConsumerOp;
+
+public class MultiOnTerminate<T> {
+
+    private final Multi<T> upstream;
+
+    public MultiOnTerminate(Multi<T> upstream) {
+        this.upstream = ParameterValidation.nonNull(upstream, "upstream");
+    }
+
+    /**
+     * Attaches an action that is executed when the {@link Multi} emits a completion or a failure or when the subscriber
+     * cancels the subscription.
+     *
+     * @param callback the consumer receiving the failure if any and a boolean indicating whether the termination
+     *        is due to a cancellation (the failure parameter would be {@code null} in this case). Must not
+     *        be {@code null}.
+     * @return the new {@link Multi}
+     */
+    public Multi<T> invoke(BiConsumer<Throwable, Boolean> callback) {
+        return Infrastructure.onMultiCreation(new MultiSignalConsumerOp<>(
```

Comment:
I would have implemented a specific operator.

Reply:
That's an option, but what would be the motivation?

Reply:
Looking at usages on `MultiSignalConsumerOp`, it's being used in `MultiOnCompletion`, `MultiOnEvent`, `MultiOnFailure`, `MultiOnItem`, and now `MultiOnTerminate`.

Reply:
Yes, I would like to avoid this. The class is unnecessary complicated because of this.

Reply:
I'll try making an alternative with a custom operator later today.

Reply:
@cescoffier added a `MultiOnTerminationInvoke` operator.